external-crates/move/crates/move-analyzer/prettier-plugin/src/index.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import * as path from 'path';
+
+import Parser = require("web-tree-sitter");
+import { SyntaxNode } from 'web-tree-sitter'
+
+import { print } from './printer'
+
+export const languages = [
+    {
+        name: 'move',
+        extensions: ['.move'],
+        parsers: ['move-parse'],
+    },
+]
+
+export const parsers = {
+    'move-parse': {
+        parse: (text: string): Promise<SyntaxNode> => {
+            return (async (): Promise<SyntaxNode> => {
+                await Parser.init();
+                const parser = new Parser();
+                const Lang = await Parser.Language.load(path.join(__dirname,
+                                                                  '..',
+                                                                  'tree-sitter-move.wasm'));
```

Comment:
nit: strange formatting, maybe we need a different prettier config for these TS files? 😄

Reply:
Admittedly, it was formatted by hand... :-) Changed it to something hopefully more palatable.